### PR TITLE
chore: Automate Admin Dashboard Build Process and Push Built Folder to Repository

### PR DESCRIPTION
**Pull request by alenmestrov**
## Summary

This PR introduces automation for the Admin Dashboard build process. When files within the `node-ui` directory are modified and committed, the Admin Dashboard is automatically built and the `build` folder is committed separately. The process ensures that the developers/users does not need to manually build Admin dashboard app when cloning the repository and starting the node.

### Key Changes:
1. **Husky Pre-commit and Post-commit Hooks**:
   - The pre-commit hook handles linting, formatting, and building.
   - The post-commit hook ensures that the `build` folder is committed in a separate commit automatically, without triggering an infinite loop.
   
2. **Commit Message Guard**:
   - The post-commit hook uses a guard condition to prevent recursion by checking the commit message for the term `"build commit"`, ensuring the build commit does not trigger the hook again.

3. **Resolved Push Issues**:
   - Increased `http.postBuffer` to handle larger pushes, ensuring smooth push of the `build` folder even for larger files or payloads.

## Details
- Post-commit hook stages and commits the `build` folder with a special commit message to avoid re-triggering itself.
- The `build` folder is committed separately to maintain a clean commit history and avoid unnecessary file changes in feature commits.
  
## How to Test

1. Modify any file in the `node-ui` directory (e.g., `.js`, `.ts`, `.tsx`).
2. Run `git commit`.
3. Verify that:
   - The post-commit hook runs the build process and commits the `build` folder separately.
4. Run `git push` and ensure the push works smoothly with the new buffer size configuration.

### CODE REVIEW

File: node-ui/src/api/httpClient.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export interface HttpClient {
   ): Promise<ResponseData<T>>;
   head(url: string, headers?: Header): Promise<ResponseData<void>>;
 }
-
+// AxiosHttpClient.ts
```
Comment (MatejVukosav) [inline node-ui/src/api/httpClient.ts:32]: what is the value of this comment?

File: .husky/post-commit [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  echo "Building the Next.js app..."
+  (cd node-ui && pnpm run build)
+
+  # Add the build folder to Git
+  echo "Adding the build folder to the Git staging area..."
+  git add node-ui/build/
+
+  # Create a separate commit for the build folder
+  echo "Creating a separate commit for the build folder..."
+  git commit -m "chore: add build folder after automatic build [build commit]" --no-verify
```
Comment (MatejVukosav) [inline .husky/post-commit]: dev should decide when he wants to push stuff

File: .husky/post-commit [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+# Check if there were changes in the 'node-ui' directory
+if git diff --name-only HEAD^ | grep -q '^node-ui/'; then
+  echo "Building the Next.js app..."
+  (cd node-ui && pnpm run build)
+
+  # Add the build folder to Git
+  echo "Adding the build folder to the Git staging area..."
+  git add node-ui/build/
```
Comment (MatejVukosav) [inline .husky/post-commit]: dashboard should be built in CI to ensure there is always the latest build

File: node-ui/src/api/httpClient.ts (hunk 1 above)
Comment (alenmestrov) [reply node-ui/src/api/httpClient.ts:32]: this was a test to see if its going to trigger build process

Comment (MatejVukosav) [reply node-ui/src/api/httpClient.ts:32]: clean?

File: .github/workflows/build_node_ui.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/checkout@v4
+
+      # Check if there are changes in the node-ui folder (excluding build)
+      - name: Check for changes in node-ui (excluding build folder)
+        id: check_changes
+        run: |
+          # Check if there are changes in the node-ui folder excluding the build directory
+          if git diff --quiet HEAD^ HEAD -- 'node-ui' ':!node-ui/build'; then
+            echo "No changes in node-ui (excluding build folder)."
+            echo "no_changes=true" >> $GITHUB_ENV
```
Comment (MatejVukosav) [inline .github/workflows/build_node_ui.yml]: exit 0

**Review by MatejVukosav — APPROVED** [review]
PR description is not in sync with the code. 

Ok as a temporary solution but as discussed, build should not be included in git.

File: node-ui/.gitignore [hunk 5]
```diff
@@ -8,7 +8,6 @@ pnpm-debug.log*
 lerna-debug.log*
 
 node_modules
```
Comment (MatejVukosav) [inline node-ui/.gitignore:10]: return build to gitignore

**Pull request merged**